### ISSUE TITLE
Fails on reuse

### ISSUE THREAD

**Issue by tarkhil** [mschout/Log-Log4perl-Layout-JSON#11]
With config 

```
log4perl.rootLogger=DEBUG, LOGFILE, LOKI
log4perl.logger.Stack.INT3.Counters.Daemon=DEBUG,LOGFILE,LOKI

log4perl.appender.LOGFILE=Log::Log4perl::Appender::File
log4perl.appender.LOGFILE.filename=/tmp/xware2.log
log4perl.appender.LOGFILE.mode=append
log4perl.appender.LOGFILE.layout=PatternLayout
log4perl.appender.LOGFILE.layout.ConversionPattern=[%d{yyyy-MM-dd HH:mm:ss.SSS}] %m{chomp} [%X{id}](%X{title}) %C %l %n

log4perl.appender.LOKI=Log::Log4perl::Appender::File
log4perl.appender.LOKI.filename=/tmp/xware.json
log4perl.appender.LOKI.layout = Log::Log4perl::Layout::JSON
log4perl.appender.LOKI.layout.field.message = %m{chomp}
log4perl.appender.LOKI.layout.field.category = %c
log4perl.appender.LOKI.layout.field.class = %C
log4perl.appender.LOKI.layout.field.fileline = %F{1}:%L
log4perl.appender.LOKI.layout.field.timestamp = %d{yyyy-MM-dd HH:mm:ss.SSS}
log4perl.appender.LOKI.layout.include_mdc = 0
log4perl.appender.LOKI.layout.utf8 = 1

log4perl.appender.LOKI2=Log::Log4perl::Appender::File
log4perl.appender.LOKI2.filename=/tmp/xware.json
log4perl.appender.LOKI2.layout = Log::Log4perl::Layout::JSON
log4perl.appender.LOKI2.layout.field.message = %m{chomp}
log4perl.appender.LOKI2.layout.field.category = %c
log4perl.appender.LOKI2.layout.field.class = %C
log4perl.appender.LOKI2.layout.field.file = %F{1}
log4perl.appender.LOKI2.layout.field.sub = %M{1}
log4perl.appender.LOKI2.layout.canonical = 1
```

Log::Log4perl::init fails with

```
Not a HASH reference at /usr/local/lib/perl5/site_perl/Log/Log4perl/Layout/JSON.pm line 174.
```

or a similar error. Replacing LOKI with LOKI2 in second line of config resolve the problem, but it seems to be wrong. 

I'm trying to figure out why reuse of appender fails, it's something with processing field values.

**Comment by tarkhil** [comment]
Sorry, I've tried to look at the code, but it's too write-only for me.

**Comment by mschout** [comment]
This should be fixed in #10, which is fixed in v0.60 which just now was uploaded to PAUSE.  Should show up in CPAN mirrors in a few hours.
